### PR TITLE
Fix icon and update logos and favicon

### PR DESCRIPTION
**Pull request by sisp**
## What changes are proposed in this pull request?

The two circular arrows in the MLflow icon do not have identical shape which I believe is unintended. The top-left arrow has an inner radius of ~7.5 whereas the bottom-right arrow has an inner radius of ~7.7.

This PR fixes the icon by making the bottom-right arrow a flipped copy of the top-left arrow. The logo is updated accordingly and all PNG images and the favicon are re-generated from the vector graphics.

Follow-up of #5529.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

## Release Notes

none

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
Can we remove `TM` in the new image?

image

File: assets/icon.svg [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <svg width="24" height="25" viewBox="0 0 24 25" fill="none" xmlns="http://www.w3.org/2000/svg">
 <path d="M18.9509 2.17634C14.0635 -1.24345 7.40926 -0.571051 3.30485 3.75735C-0.799556 8.08576 -1.11776 14.7663 2.55667 19.4652L6.22524 16.7724C4.40423 14.5134 4.03437 11.4124 5.27298 8.78843C6.51159 6.16448 9.1408 4.47914 12.0422 4.44929L11.9554 7.31581L18.9509 2.17634Z" fill="#43C9ED"/>
-<path d="M21.6639 4.85176C21.5423 4.68382 21.4149 4.51878 21.2846 4.35953L17.7521 6.96546C19.7463 9.17232 20.2637 12.3424 19.0746 15.0688C17.8856 17.7952 15.2104 19.5728 12.2362 19.6129L12.3231 16.7493L5.24945 21.9611C10.0971 25.2255 16.5896 24.5337 20.6406 20.3213C24.6917 16.1089 25.1295 9.59437 21.6784 4.87782L21.6639 4.85176Z" fill="#0194E2"/>
+<path d="M5.0491 21.8237C9.9365 25.2435 16.5907 24.5711 20.6951 20.2427C24.7996 15.9142 25.1178 9.2337 21.4433 4.5348L17.7748 7.2276C19.5958 9.4866 19.9656 12.5876 18.727 15.2116C17.4884 17.8355 14.8592 19.5209 11.9578 19.5507L12.0446 16.6842L5.0491 21.8237Z" fill="#0194E2"/>
```
Comment (harupy) [inline assets/icon.svg:3]: Is it possible to visualize the difference?

**Comment by sisp** [comment]
No, the `TM` wasn't added on purpose. It's in the SVG logo and I simply exported a PNG version, that's why it was added. I've pushed a commit where it's removed again.

**Comment by sisp** [comment]
Ah, would you like the `TM` removed in the SVG logo, too? I merely committed the unmodified SVGs in #5529 and the `TM` was included there.

**Comment by harupy** [comment]
@sisp Thanks for the reply. We discussed whether the logo should include the TM. Let's add it. Sorry for the confusion.

**Review by harupy — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/5618#issuecomment-1090964418 is addressed!

**Comment by sisp** [comment]
@harupy No worries, I've reverted the last commit, so the `TM` is included in the PNG again.

**Comment by harupy** [comment]
thanks!

**Pull request merged**